torch/csrc/mps/Module.cpp

```diff
@@ -0,0 +1,85 @@
+#include <ATen/ATen.h>
+#include <ATen/mps/MPSAllocatorInterface.h>
+#include <ATen/mps/MPSGeneratorImpl.h>
+
+#include <torch/csrc/Generator.h>
+#include <torch/csrc/python_headers.h>
+#include <torch/csrc/utils/pybind.h>
+#include <torch/csrc/utils/python_numbers.h>
+#include <torch/csrc/utils/python_strings.h>
+
+using namespace torch;
+
+static PyObject* MPSModule_initExtension(PyObject* self, PyObject* noargs) {
+#if C10_ASAN_ENABLED
+  TORCH_WARN(
```

Comment:
> This needs to be inside the HANDLE_TH_ERRORS macro otherwise it won't be translated to a python warning properly.

Code is taken from CUDA. The `HANDLE_TH_ERRORS` is at line 21 below.

> Also this is going to be triggered as soon as the user import torch

This is meant to be run on-demand only from `lazy_init()` via `torch._C._mps_init()`. By on-demand, I mean only when user for example calls `torch.mps.manual_seed()`, then if the module isn't initialized already, then it'll initialize it lazily.

Reply:
> Code is taken from CUDA. The HANDLE_TH_ERRORS is at line 21 below.

Well the CUDA code is wrong and should be changed :(

Reply:
Removed all the redundant init code.